plugins/com.google.cloud.tools.eclipse.projectselector.test/src/com/google/cloud/tools/eclipse/projectselector/MiniSelectorTest.java

```diff
@@ -89,31 +101,45 @@ public void testListenerFired() {
 
     assertNotNull(selector.getSelection());
     assertTrue(selector.getSelection().isEmpty());
-    final boolean[] calledAndCorrect = new boolean[] {false};
+    final AtomicReference<ISelection> lastSelection = new AtomicReference<>();
     selector.addSelectionChangedListener(new ISelectionChangedListener() {
       @Override
       public void selectionChanged(SelectionChangedEvent event) {
-        assertThat(event.getSelection(), instanceOf(IStructuredSelection.class));
-        IStructuredSelection selection = (IStructuredSelection) event.getSelection();
-        assertEquals(1, selection.size());
-        assertThat(selection.getFirstElement(), instanceOf(GcpProject.class));
-        GcpProject gcpProject = (GcpProject) selection.getFirstElement();
-        calledAndCorrect[0] =
-            "foo".equals(gcpProject.getName()) && "foo.id".equals(gcpProject.getId());
+        lastSelection.set(event.getSelection());
```

Comment:
This should be happening only in the UI thread. Do we need `AtomicReference`?

Reply:
It's just meant for a convenient reference holder.  `SoftReference` and `WeakReference`  may lose the reference.  `final ISelection[] lastSelection = new ISelection[1];`.  I'll change this.